syntax="proto3";

message TritonPredictRequest{
    bytes chunk = 1;
}

message TritonPredictResponse{
    string classification = 1;
    string details = 2;
}

service TritonPredictService{
    rpc TritonPredict(TritonPredictRequest) returns (TritonPredictResponse);
}

message UploadVideoRequest{
    bytes chunk = 1;
}

message UploadVideoResponse{
    repeated TritonPredictResponse predictions = 1;
}

service UploadVideoService{
    rpc UploadVideo(UploadVideoRequest) returns (UploadVideoResponse);
}

message UploadImageRequest{
    bytes chunk = 1;
}

message UploadImageResponse{
    TritonPredictResponse prediction = 1;
}

service UploadImageService{
    rpc UploadImage(UploadImageRequest) returns (UploadImageResponse);
}

